### PR TITLE
Remove unnecessary code

### PR DESCRIPTION
**Pull request by Forgind**
Removes UESanitized as discussed in #4899.

Also converts some repetitive tests to theories.

### CODE REVIEW

File: src/MSBuild.UnitTests/CommandLineSwitches_Tests.cs [hunk 1]
```diff
@@ -49,205 +49,114 @@ public void BogusSwitchIdentificationTests()
...
-            Assert.True(CommandLineSwitches.IsParameterlessSwitch("h", out parameterlessSwitch, out duplicateSwitchErrorMessage));
-            Assert.Equal(CommandLineSwitches.ParameterlessSwitch.Help, parameterlessSwitch);
-            Assert.Null(duplicateSwitchErrorMessage);
-
-            Assert.True(CommandLineSwitches.IsParameterlessSwitch("H", out parameterlessSwitch, out duplicateSwitchErrorMessage));
-            Assert.Equal(CommandLineSwitches.ParameterlessSwitch.Help, parameterlessSwitch);
-            Assert.Null(duplicateSwitchErrorMessage);
-
-            Assert.True(CommandLineSwitches.IsParameterlessSwitch("?", out parameterlessSwitch, out duplicateSwitchErrorMessage));
+            Assert.True(CommandLineSwitches.IsParameterlessSwitch(help, out CommandLineSwitches.ParameterlessSwitch parameterlessSwitch, out string duplicateSwitchErrorMessage));
```
Comment (BenVillalobos) [inline src/MSBuild.UnitTests/CommandLineSwitches_Tests.cs]:
```suggestion
            CommandLineSwitches.IsParameterlessSwitch(help, out CommandLineSwitches.ParameterlessSwitch parameterlessSwitch, out string duplicateSwitchErrorMessage).ShouldBeTrue();
```

While you're switching over to using InlineData, we might as well start using Shouldly here. This applies to each test.

Comment (Forgind) [reply src/MSBuild.UnitTests/CommandLineSwitches_Tests.cs]: I did it for all the tests that I switched to theories. The others have a different style, so they'd take longer.

File: src/MSBuild.UnitTests/CommandLineSwitches_Tests.cs [hunk 2]
```diff
@@ -362,44 +238,21 @@ public void IgnoreProjectExtensionsSwitchIdentificationTests()
             bool unquoteParameters;
             bool emptyParametersAllowed;
 
-            Assert.True(CommandLineSwitches.IsParameterizedSwitch("ignoreprojectextensions", out parameterizedSwitch, out duplicateSwitchErrorMessage, out multipleParametersAllowed, out missingParametersErrorMessage, out unquoteParameters, out emptyParametersAllowed));
+            Assert.True(CommandLineSwitches.IsParameterizedSwitch(ignoreprojectextensions, out parameterizedSwitch, out duplicateSwitchErrorMessage, out multipleParametersAllowed, out missingParametersErrorMessage, out unquoteParameters, out emptyParametersAllowed));
```
Comment (BenVillalobos) [inline src/MSBuild.UnitTests/CommandLineSwitches_Tests.cs]: ;)
```suggestion
CommandLineSwitches.IsParameterizedSwitch(ignoreprojectextensions, out parameterizedSwitch, out duplicateSwitchErrorMessage, out multipleParametersAllowed, out missingParametersErrorMessage, out unquoteParameters, out emptyParametersAllowed).ShouldBeTrue();
```

File: src/MSBuild.UnitTests/CommandLineSwitches_Tests.cs [hunk 3]
```diff
@@ -362,44 +238,21 @@ public void IgnoreProjectExtensionsSwitchIdentificationTests()
...
             Assert.Equal(CommandLineSwitches.ParameterizedSwitch.IgnoreProjectExtensions, parameterizedSwitch);
-            Assert.Null(duplicateSwitchErrorMessage);
+            duplicateSwitchErrorMessage.ShouldBeNull();
             Assert.True(multipleParametersAllowed);
```
Comment (BenVillalobos) [inline src/MSBuild.UnitTests/CommandLineSwitches_Tests.cs]: While we're at it, shouldy-ify the rest of the Assert.X's

**Pull request merged**